tecken/dockerflow_extra.py

```diff
@@ -55,7 +55,7 @@ def check_url(url, setting_key):
         if url in checked:
             return
         bucket = StorageBucket(url)
-        if not bucket.private:
+        if bucket.is_emulated_gcs or not bucket.private:
```

Comment:
Why this check here? Do we do the same for minio?

Also, since private bucket has nothing to do with emulated environment, I'd split this up and add a comment.

```
if bucket.is_emulated_gcs:
    # something something exception with emulated environments
    return
if bucket.private:
    return
```

Reply:
I want to verify that this is needed, it is possible that something broke (tests? bootstrapping?) before this change, but that a later fix resolved it.

While I'm doing that, do you know why do we skip verifying the existence of private buckets?

Reply:
I can't think of any reason why we'd want to verify urls for private buckets, but not public ones. That's puzzling.

Reply:
oops, it is the other way around. This verifies public but not private buckets. I'm guessing that doesn't make it any clearer why.

Reply:
The code is able to verify buckets in the GCS emulator. I must have added this before the automatic bucket creation code.

I'll remove the change, which will leave ``dockerflow_extra.py`` unchanged for this PR, and the skipped private bucket checks as a mystery.